plugins/keysend.c

```diff
@@ -171,13 +171,19 @@ static struct command_result *json_keysend(struct command *cmd, const char *buf,
 	p->deadline = timeabs_add(time_now(), time_from_sec(*retryfor));
 	p->getroute->riskfactorppm = 10000000;
 
+	if (node_id_eq(&my_id, p->destination)) {
+		return command_fail(
+		    cmd, JSONRPC2_INVALID_PARAMS,
+		    "We are the destination. Keysend cannot be used to send funds to yourself");
+	}
+
 	if (!amount_msat_fee(&p->constraints.fee_budget, p->amount, 0,
 			     *maxfee_pct_millionths / 100)) {
-		tal_free(p);
```

Comment:
Notice that this is allocated off of `cmd`, which we free at the end of the JSON-RPC call anyway. So no need to free it early